app/connectors/connectors/rest/rest-swagger-connector/src/main/java/io/syndesis/connector/swagger/SwaggerConnectorComponent.java

```diff
@@ -40,24 +40,73 @@
 
 public class SwaggerConnectorComponent extends DefaultConnectorComponent {
 
+    private String accessToken;
+
+    private AuthenticationType authenticationType;
+
     private String specification;
 
     public SwaggerConnectorComponent() {
         this(null);
     }
 
     public SwaggerConnectorComponent(final String componentSchema) {
-        super("swagger-connector", componentSchema, SwaggerConnectorComponent.class.getName());
+        super("swagger-connector", componentSchema, SwaggerConnectorComponent.class);
+    }
+
+    public String getAccessToken() {
+        return accessToken;
+    }
+
+    public AuthenticationType getAuthenticationType() {
+        return authenticationType;
     }
 
     public String getSpecification() {
         return specification;
     }
 
+    public void setAccessToken(final String accessToken) {
+        this.accessToken = accessToken;
+    }
+
+    public void setAuthenticationType(final AuthenticationType authenticationType) {
+        this.authenticationType = authenticationType;
+    }
+
     public void setSpecification(final String specification) {
         this.specification = specification;
     }
 
+    /* default */ void addAuthenticationHeadersTo(final Map<String, Object> headers) {
+        if (authenticationType == AuthenticationType.oauth2) {
```

Comment:
Normally I'd prefer to add a `AuthenticationType.addToHeaders()` methods which avoid such if-else checks. But I'm fine with this (i.e. if authenticationType can  be null)

Reply:
I think that way would be nice when we add the third way of authenticating (#436)